eng/pipelines/templates/Apex_Tests_On_Windows.yml

```diff
@@ -1,115 +1,83 @@
-
-- task: MSBuild@1
-  displayName: "Restore Apex Tests"
-  inputs:
-    solution: "build\\build.proj"
-    configuration: "$(BuildConfiguration)"
-    # Set MSBuildEnableWorkloadResolver to work around https://github.com/dotnet/sdk/issues/17461
-    msbuildArguments: "/t:RestoreApex /p:BuildNumber=$(BuildNumber) /p:MSBuildEnableWorkloadResolver=false"
-
-- task: MSBuild@1
-  displayName: "Run Apex Tests (continue on error)"
-  timeoutInMinutes: 120
-  continueOnError: "true"
-  inputs:
-    solution: "build\\build.proj"
-    configuration: "$(BuildConfiguration)"
-    # Set MSBuildEnableWorkloadResolver to work around https://github.com/dotnet/sdk/issues/17461
-    msbuildArguments: "/t:ApexTestsStandalone /p:TestResultOutputFormat=xml /p:BuildNumber=$(BuildNumber) /p:MSBuildEnableWorkloadResolver=false"
-    maximumCpuCount: true
-  condition: "and(succeeded(), eq(variables['IsOfficialBuild'], 'true'))"
-
-- task: MSBuild@1
-  displayName: "Run Apex Tests (stop on error)"
-  timeoutInMinutes: 120
-  continueOnError: "false"
-  inputs:
-    solution: "build\\build.proj"
-    configuration: "$(BuildConfiguration)"
-    # Set MSBuildEnableWorkloadResolver to work around https://github.com/dotnet/sdk/issues/17461
-    msbuildArguments: "/t:ApexTestsStandalone /p:TestResultOutputFormat=xml /p:BuildNumber=$(BuildNumber) /p:MSBuildEnableWorkloadResolver=false"
-    maximumCpuCount: true
-  condition: "and(succeeded(), not(eq(variables['IsOfficialBuild'], 'true')))"
-
-- task: PublishTestResults@2
-  displayName: "Publish Test Results"
-  inputs:
-    testRunner: "XUnit"
-    testResultsFiles: "*.xml"
-    searchFolder: "$(System.DefaultWorkingDirectory)\\build\\TestResults"
-    mergeTestResults: "true"
-    testRunTitle: "NuGet.Client Apex Tests On Windows"
-  condition: "succeededOrFailed()"
-
-- task: PowerShell@1
-  displayName: "Initialize Git Commit Status on GitHub"
-  inputs:
-    scriptType: "inlineScript"
-    inlineScript: |
-      . $(Build.Repository.LocalPath)\\scripts\\utils\\PostGitCommitStatus.ps1
-      SetCommitStatusForTestResult -PersonalAccessToken $(NuGetLurkerPersonalAccessToken) -VstsPersonalAccessToken $(System.AccessToken) -CommitSha $(Build.SourceVersion) -TestName "Apex Tests On Windows"
-  condition: "not(eq(variables['ManualGitHubChecks'], 'false'))"
-
-- task: PowerShell@1
-  displayName: "Kill running instances of DevEnv"
-  inputs:
-    scriptType: "inlineScript"
-    inlineScript: |
-      . $(Build.Repository.LocalPath)\\scripts\\e2etests\\VSUtils.ps1
-      KillRunningInstancesOfVS (Get-LatestVSInstance -VersionRange (Get-VisualStudioVersionRangeFromConfig))
-  condition: "always()"
+        - task: PowerShell@1
+          displayName: "SetupFunctionalTests.ps1"
+          inputs:
+            scriptName: "$(System.DefaultWorkingDirectory)/artifacts/EndToEnd/scripts/SetupFunctionalTests.ps1"
+      postDeployAndRunTestsStepList:
+        - task: PowerShell@1
```

Comment:
Previously there was publish test result step. Is it not necessary anymore?

Reply:
AFAIK, DartLab publishes test results by default and hence I didn't include that as a separate step. For e.g. a test failed once on [CI](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5264760&view=ms.vss-test-web.build-test-results-tab) for the same private branch and when I navigated to the `Tests` tab, I saw all the test failures along with other information without having an additional step to publish test results.

![image](https://user-images.githubusercontent.com/52756182/136296310-0a94fbc9-40d2-4f29-a931-6c7762267ad9.png)

Reply:
I see.  Where is pumping of tests into Dartlab? What is actually instructing Dartlab to run Apex tests?

Reply:
Andy did all the challenging work to inject `NuGet Apex tests` binaries into DartLab drop in the previous PR https://github.com/NuGet/NuGet.Client/blob/dev/eng/pipelines/templates/Build_and_UnitTest.yml#L356

Please navigate to `Apex Tests on Windows -> Test Execution -> Run Tests` step which is running the tests using the parameters such as `runSettingsUri` file and apex test binaries passed to it.